tests/queries/tests.py

```diff
@@ -4549,3 +4549,49 @@ def test_ticket_23622(self):
             set(Ticket23605A.objects.filter(qy).values_list("pk", flat=True)),
         )
         self.assertSequenceEqual(Ticket23605A.objects.filter(qx), [a2])
+
+
+class TestComment(TestCase):
```

Comment:
Can we add a comment for combined queries:
```python
qs1 = NamedCategory.objects.comment("First comment")
qs2 = NamedCategory.objects.comment("Second comment")
q1.union(qs2).comment("Main comment")
```

Reply:
Interesting example! Haven't looked into the implementation of `union` in detail but it feels like the `q1` queryset is compiled when `union` is called thus missing to add "Main comment"  but `self.query.comments` contains both of them when the actual union is compiled. The question is how we can/should distinguish between a comment added on the plain queryset and the "combinator" queryset.